packages/vm/src/modules/pinned_memory_cache.rs

```diff
@@ -4,9 +4,19 @@ use std::collections::HashMap;
 use super::cached_module::CachedModule;
 use crate::VmResult;
 
+/// Struct storing some additional metadata, which is only of interest for the pinned cache,
+/// alongside the cached module.
+// TODO: Maybe implement a `Deref` for this? But would it even worth it considering how little it is actually used?
+pub struct InstrumentedModule {
```

Comment:
I think this is a decent implementation. An unnamed tuple would probably work as well.

But I would not implement Deref as this has historically been overused and should be reserved for smart pointers.